apps/web/src/pages/templates/components/EditorSidebarHeaderActions.tsx

```diff
@@ -38,17 +39,26 @@ export const EditorSidebarHeaderActions = () => {
   const hasNoFilters = (filters && filters?.length === 0) || !filters;
 
   const onAddVariant = () => {
-    const newVariant = addVariant(stepUuid);
-    if (newVariant) {
-      // TODO: show the conditions sidebar first and then when conditions are applied show the variant editor
-      navigate(basePath + `/${newVariant?.template.type}/${stepUuid}/variants/${newVariant.uuid}`);
+    const variant = addVariant(stepUuid);
+    if (variant) {
+      navigate(basePath + `/${variant?.template.type}/${stepUuid}/variants/${variant?.uuid}/conditions`);
```

Comment:
@BiswaViraj what if after navigation to conditions, user clicks cancel. As the variant was already "created" inside the form, wouldn't it be updated as well when updating the workflow?

Reply:
It redirects to the variant editor.
Yes it will create the variant. Without creating the variant it will be difficult to display the conditions screen, as we do show the variant name and also apply the conditions

Reply:
I do agree with Gali. Maybe we need to have a path like: `/${variant?.template.type}/${stepUuid}/variants/new/conditions` and create the variant on the conditions apply button click?

Reply:
We can do that, just thinking how we will handle the Conditions name on the header

Reply:
The variant is a copy of the root step, so we can take a title from the root and add the prefix `V1` in the beginning, like the same way we create the variant name.